source/common/network/transport_socket_options_impl.cc

```diff
@@ -77,9 +79,15 @@ TransportSocketOptionsConstSharedPtr TransportSocketOptionsUtility::fromFilterSt
     proxy_protocol_options.emplace(typed_data->value());
   }
 
+  if (auto typed_data =
+          filter_state->getDataReadOnly<FilterStateProxyInfo>(FilterStateProxyInfo::key());
+      typed_data != nullptr) {
+    proxy_info = std::make_unique<TransportSocketOptions::ProxyInfo>("host", typed_data->address());
```

Comment:
Ah, since this is passing in `"host"` I'm guessing it's an unused argument?

Reply:
Suggestion to use well-known endpoint metadata ("tunnel.address" or something). We've been doing that for the internal listener effectively.

Reply:
using the hostname() sorry for missing this as I fixed things up.